internal/controller/apicerts/retrieve_from_secret.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package apicerts
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+type RetrieveFromSecretFunc func(secret *corev1.Secret) ([]byte, []byte)
```

Comment:
Should this be split into separate types? Confirm that the call sites only ever want a CA or a serving cert.